packaging/package/jar_jdk_signer.py

```diff
@@ -0,0 +1,66 @@
+import os
+import logging
+import subprocess
+from pathlib import Path
+from .helper import check_jdk_environ_variable
+
+JARSIGNER_EXECUTABLE_NAME = "jarsigner.exe"
+logger = logging.getLogger(__name__)
+
+
+def validate_signing_input(input_dir, taco_name, alias, keystore):
+    """
+    Validate signing input
+    """
+
+    if not alias:
+        logger.error("Signing Error: Private key's alias is missing or empty")
+        return False
+
+    if not keystore or not os.path.isfile(Path(keystore)):
+        logger.error("Signing Error: Keystore path is missing or invalid")
+        return False
+
+    if not os.path.isfile(input_dir/taco_name):
+        logger.error("Signing Error: Taco file to be signed has been deleted or doesn't exist")
```

Comment:
do you also want to check that the extension is .taco ?

Reply:
Since we can only do package and then sign, the input for signing is the output of 
packaging. see from package_name variable line 49 to line 62 in package.py.